File: src/SqlStudio.Core/Models/SavedConnection.cs
namespace SqlStudio.Core.Models;

public class SavedConnection
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Server { get; set; } = string.Empty;
    public string Database { get; set; } = "master";
    public ConnectionAuthType AuthType { get; set; }
    public string? Username { get; set; }
    public string? EncryptedPassword { get; set; }
    public string? TenantId { get; set; }
    public bool TrustServerCertificate { get; set; }
    public bool Encrypt { get; set; } = true;
    public int Port { get; set; } = 1433;
    public DateTime LastConnected { get; set; }
    public Guid? GroupId { get; set; }
    public string? EnvironmentLabel { get; set; }
}

File: SQLExplorer/Models/SavedConnection.swift
import Foundation

struct SavedConnection: Identifiable, Codable {
    var id: UUID = UUID()
    var name: String = ""
    var server: String = ""
    var port: Int = 1433
    var database: String = "master"
    var authType: ConnectionAuthType = .entraIdInteractive
    var username: String?
    var tenantId: String?
    var trustServerCertificate: Bool = true
    var encrypt: Bool = true
    var lastConnected: Date = Date()
    var groupId: UUID?
    var environmentLabel: String?
}
